typing/ctype.ml

```diff
@@ -835,7 +836,7 @@ let rec update_level env level expand ty =
         set_level ty level;
         iter_type_expr (update_level env level expand) ty
     | Tfield(lab, _, ty1, _)
-      when lab = dummy_method && (repr ty1).level > level ->
+      when lab = dummy_method && level < (repr ty1).scope ->
         raise Trace.(Unify [escape Self])
```

Comment:
Good idea. The prohibition to lower the level of the dummy method (even locally) was a real pain.